### PR TITLE
Relocate doc search bar

### PR DESCRIPTION
**Pull request by magyarn**
This PR addresses issue #799 

Moving the search bar to be in-line with the "Documentation" header makes it more readily accessible when a user arrives on any documentation page (previously he/she had to scroll to near the bottom to use it).

The previous placeholder text also turned out to be too long on mobile devices (it overflowed beyond the width of the search bar, appearing behind the search button icon). The previous text displayed the version of documentation a user was viewing/could search, but since this information is already displayed in the bottom right corner version button, we can safely remove it and simply use "Search" as the new, shorter placeholder text.

### CODE REVIEW

**Comment by m-aciek** [comment]
@magyarn Could you provide screenshots to show "before" and "after" of your change?

**Comment by pauloxnet** [comment]
# Screenshots

I took these images with responsive mode browser at resolution of 768x432 px (16/9) with vertical and horizontal rotation to have the desktop and mobile version of the same page.

## Disclaimer

_I'm answering to @m-aciek request because I've opened the original issue #799 and because I've worked together with @magyarn to this PR at the DjangoCon Europe 2019 sprints._

## Desktop version

### Before

The search input now is quite a the top of the page but is in the right column and it is not a thing you see clearly at first look.

Screen Shot 2019-04-17 at 08 57 57

### After

In this implementation the search input is a relevant element of the page and it's clear you are searching only in the documentation as the input is at the right of the title.

Screen Shot 2019-04-17 at 09 05 05

## Mobile version

### Before

As you see the search input is at the bottom of the long page and you need to scroll down every time you change page to do a new search.

docs djangoproject com_en_2 2_

### After

In the new implementation the search input is at the top of the page and it's easy to use every time you change page.

Screen Shot 2019-04-17 at 09 05 09

## Search results

### Before

Now you have to scrool down tot the bottom of the page to do a search

Screen Shot 2019-04-17 at 09 05 47

In the search result page the search input is still a the bottom of the page of your results.

Screen Shot 2019-04-17 at 09 06 01

## After

The input search is at the top of your page before and after the search execution so it's possible to do another search easily if it's necessary.

Screen Shot 2019-04-17 at 09 05 55

**Comment by pauloxnet** [comment]
@carltongibson @felixxm can I ask you to review this PR we sent during the DjangoCon Europe 2019 Sprint day ? Thank you!

**Comment by rixx** [comment]
This looks really good!

**Comment by pauloxnet** [comment]
@carltongibson I know there's a lot of work with Django 3.0 but I hope you can review and merge this PR before it becomes obsolete. Thanks.

**Comment by pauloxnet** [comment]
Thanks for your review, I hope @magyarn can update the PR based on your suggestions.

**Comment by magyarn** [comment]
Hi @carltongibson and @pauloxnet, I totally agree! I've updated the PR per your suggestions. Attached a screenshot of the expected rendering now. 

Screen Shot 2019-09-13 at 8 35 12 AM

**Comment by rixx** [comment]
This looks wonderful – thank you for adapting your PR!

File: djangoproject/scss/_style.scss [hunk 1]
```diff
@@ -412,6 +412,12 @@ blockquote {
     align-items: center;
     justify-content: space-between;
 }
+.container--flex--wrap--mobile {
+    flex-wrap: wrap;
+    @include respond-min(768px) {
+        flex-wra-: no-wrap;
```
Comment (carltongibson) [inline djangoproject/scss/_style.scss]: Typo?

File: docs/forms.py [hunk 2]
```diff
@@ -12,5 +12,5 @@ def __init__(self, data=None, **kwargs):
         super().__init__(data=data, **kwargs)
         self.fields['q'].widget = forms.TextInput(attrs={
             'type': 'search',
-            'placeholder': search_placeholder % self.release.human_version
+            'placeholder': search_placeholder.format(self.release.version)
```
Comment (carltongibson) [inline docs/forms.py]: I think just revert these two changes no?

**Comment by magyarn** [comment]
Hi @pauloxnet and @carltongibson, I made updated the PR and tried to add/commit just the two modified files but I fear those changes caused a CI issue? Do you know what could be going on?

File: docs/forms.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from django import forms
 from django.utils.translation import ugettext_lazy as _
 
-search_placeholder = _('Search %s documentation')
+search_placeholder = _('Search {} documentation')
```
Comment (pauloxnet) [inline docs/forms.py]: @magyarn I think you can try to revert this change to fix the broken tests

**Comment by magyarn** [comment]
Looks like this is good to go, @carltongibson

**Comment by carltongibson** [comment]
@magyarn Welcome aboard! 💃

**Comment by pauloxnet** [comment]
I'm happy to see this PR merged at the end.
I opened the issue #799 2 years ago and then we worked to fix it during the sprint day at DjangoCon Europe 2019 in Cophenagen 
Thanks to @carltongibson and @magyarn

**Comment by magyarn** [comment]
Thank you @pauloxnet and @carltongibson! 'Excited to have made my first contribution to the Django community!
